src/components/video-container/Video-container.component.ts

```diff
@@ -295,7 +323,63 @@ export class VideoContainer extends LitElement {
     this.togglePlay()
   }
 
+  @listen(Types.Command.setVolume)
+  @listen(Types.Command.setQualityLevel)
+  @listen(Types.Command.mute)
+  @listen(Types.Command.unmute)
+  @listen(Types.Command.toggleMuted)
+  @listen(Types.Command.setPlaybackRate)
+  @listen(Types.Command.enableTextTrack)
+  _syncStateWithStorage(params: CommandParams, _: any, command: Types.Command) {
+    if (!this._storageProvider) return
+
+    let key: keyof State
+    let value: unknown
+
+    switch (command) {
+      case Types.Command.setVolume:
+        key = 'volume'
+        value = +params.volume
+        break
+      case Types.Command.setQualityLevel:
+        key = 'activeQualityLevel'
+        value = +params.level
+        break
+      case Types.Command.toggleMuted:
+      case Types.Command.mute:
+      case Types.Command.unmute:
+        key = 'isMuted'
+        value = this.videos[0].muted
+        break
+      case Types.Command.setPlaybackRate:
+        key = 'playbackRate'
+        value = +params.playbackRate
+        break
+      case Types.Command.enableTextTrack:
+        key = 'activeTextTrack'
+        value = params.lang
+        break
+    }
+    const currentVal = this._storageProvider.get()
+    this._storageProvider.set({ ...currentVal, [key]: value })
+  }
+
   setup() {
+    // active quality level will be initialized in HLS callback
+    const { activeQualityLevel, ...savedSettings } = this._storageProvider ? this._storageProvider.get() : {} as StorageValue
```

Comment:
seems a bit weird, let's incapsulate this logic in the StoreageProvider?

Reply:
for me looks quite elegant :) 
You want to have ability to specify list of settings that should be returned by `get` method ?

Reply:
In case there is no storage key, the provider will always return an empty hash?